### PR TITLE
tests(refactor): Extract mail account management tests from `tests.bats`

### PR DESCRIPTION
**Pull request by polarathene**
# Description

Test cases should remain mostly the same, except:
- In the `user4` test, there was an account removal at the end that was probably a typo and intended as `user4` as a clean-up step ([original PR](https://github.com/docker-mailserver/docker-mailserver/pull/441)). I've dropped it as it doesn't seem necessary.
- The account password update check was part of a [PR introducing the feature](https://github.com/docker-mailserver/docker-mailserver/pull/413), but the `sleep 2` lines don't seem necessary so I've removed those.
- Two tests without `postfix-accounts.cf` via an alternative volume and `docker run` were refactored to not need that a all.
- `tests.bats` had two groups of "accounts" tests, I've separated them with some comment headers. The very first test didn't seem as thorough, and improving on that seems to have raised a bug? (_not entirely sure, just that it's related to the Dovecot dummy alias workaround, potentially related feature request https://github.com/docker-mailserver/docker-mailserver/issues/3022 _)

To assist with review, if helpful I've staged changes out into multiple commits improved diffs:
- First commit handles cut / paste between files.
- Next commit revises test case descriptions.
- Followed by refactoring of test cases.

Some test cases could probably be collapsed or changed further, but this is good enough as an initial pass :+1: 

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: test/tests/parallel/set3/mta/account_management.bats [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  # Waits until change event has created directory but not completed:
+  _wait_until_account_maildir_exists 'user3@domain.tld'
+  # Should trigger a new change event:
+  _exec_in_container setup email del -y 'user3@domain.tld'
+
+  # NOTE: This is only checking `postfix-accounts.cf`, account may still persist
+  # elsewhere momentarily such as the Dovecot UserDB until change event kicks in.
+  __check_mail_account_exists 'user3@domain.tld'
+  assert_failure
+  assert_output ''
```
Comment (georglauterbach) [inline test/tests/parallel/set3/mta/account_management.bats:97]: The command fails, and we check for empty output. Is that intended?

**Review by georglauterbach — CHANGES_REQUESTED** [review]
We should take care of the parsing of `ls` and the directory checks; the rest looks very good!

File: test/tests/parallel/set3/mta/account_management.bats (hunk 1 above)
Comment (polarathene) [reply test/tests/parallel/set3/mta/account_management.bats:98]: I mostly ported over whatever was in the original test cases. This used to be something like a `[[ -z ${output} ]]` I think. I haven't looked into if it was meant to ensure there wasn't a failure with output.

I can remove the check if it doesn't make sense?

I do recall that if the sed + grep commands didn't handle preventing the substring or `.` wildcard matches, you'd get the wrong result in either of these `__check_mail_account_exists()` calls with output. Although with `assert_failure` I guess this 2nd assert for output doesn't contribute anything? :thinking: 

```suggestion
  assert_failure
```

File: test/tests/parallel/set3/mta/account_management.bats [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  _should_output_number_of_lines 7
+
+  # Relevant log output from scripts/helpers/accounts.sh:_create_dovecot_alias_dummy_accounts():
+  # [  DEBUG  ]  Adding alias 'alias1@localhost.localdomain' for user 'user1@localhost.localdomain' to Dovecot's userdb
+  # [  DEBUG  ]  Alias 'alias2@localhost.localdomain' is non-local (or mapped to a non-existing account) and will not be added to Dovecot's userdb
+  # [  DEBUG  ]  Adding alias '@localdomain2.com' for user 'user1@localhost.localdomain' to Dovecot's userdb
+}
+
+@test "should have created maildir for 'user1@localhost.localdomain'" {
+  _run_in_container ls -d '/var/mail/localhost.localdomain/user1'
```
[the remark this answers is not in the document]
Comment (polarathene) [reply test/tests/parallel/set3/mta/account_management.bats]: It does look like that was there intent. Applied your suggestions :+1:

**Review by georglauterbach — APPROVED** [review]

**Review by casperklein — APPROVED** [review]

**Pull request merged**